test/Constraints/array_literal.swift

```diff
@@ -155,6 +155,15 @@ func defaultToAny(i: Int, s: String) {
   let _: Int = a6 // expected-error{{value of type '[A]'}}
 }
 
+func noInferAny(iob: inout B, ioc: inout C) {
+  var b = B()
+  var c = C()
+  let _ = [b, c, iob, ioc] // do not infer [Any] when elements are lvalues or inout
```

Comment:
FWIW they can't actually be inout; you still get an lvalue type when you reference an inout parameter